DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionDataDelegateSwizzler.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// Swizzles `URLSessionDataDelegate` callbacks.
+internal class URLSessionDataDelegateSwizzler {
+    static var didReceiveMap: [String: DidReceive?] = [:]
+    static var lock = NSRecursiveLock()
```

Comment:
/suggestion We have repetitive use of recursive lock now, so we could create a `@propertyWrapper` in a similar fashion as `ReadWriteLock`. It could be a `Mutex` using `pthread_mutex_t` with optional parameters such as `PTHREAD_MUTEX_RECURSIVE`.
Motivation being that [`NSLock` are slow compared to mutex](https://www.vadimbulavin.com/benchmarking-locking-apis/)

Reply:
Let me check on this.

Reply:
It sounded like premature optimization to me but I still updated the code with `pthread_mutex_t`

Reply:
Actually reverted back to original - xcode complains in ThreadSanitizer and also CI run failed